content/en/docs/concepts/workloads/controllers/job.md

```diff
@@ -867,6 +867,53 @@ is disabled, `.spec.completions` is immutable.
 Use cases for elastic Indexed Jobs include batch workloads which require 
 scaling an indexed Job, such as MPI, Horovord, Ray, and PyTorch training jobs.
 
+### Recreation only when Pod has been fully terminated
+
+{{< feature-state for_k8s_version="v1.28" state="alpha" >}}
+
+By default, the Job controller recreates pods as soon they are either failed or have a deletion timestamp.
+
+This means, that you may have two running at the same time, both corresponding to the same pod. In that case, one is terminating, while the other is starting up.
+
+This behavior might be problematic in some situations, including:
+
+- when using a framework which requires only a single pod per index (for example Tensorflow, JAX). See [issue](https://github.com/kubernetes/kubernetes/issues/115844) for more details.
+- when using a cluster autoscaler, which could increase the size of the cluster for only a short while during pod termination.
+
+If this behavior is problematic for your Jobs, you may choose to create replacement pods only when the terminating pod is fully terminal. To do this, set the `.spec.podReplacementPolicy: Failed`.
+When the `JobPodReplacementPolicy`[feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is set, you will be able to specify `PodReplacementPolicy` to `Failed`.  
+This will only recreate pods once they are terminated.  
+Default behavior will be to recreate upon deletion (`DeletionTimestamp != nil`).
+
+```yaml
+kind: Job
+metadata:
+  name: new
+  ...
+spec:
+  podReplacementPolicy: Failed
+  ...
+```
+
+When using the feature gate `JobPodReplacementPolicy`, you can inspect a new field in the JobStatus called `terminating`.  
+This will report the number pods that are currently terminating and is easily viewable in the status.
+
+```shell
+kubectl get jobs/myjob -o yaml
+```
+
+```yaml
+apiVersion: batch/v1
+kind: Job
+# .metadata and .spec omitted
+status:
+  terminating: 1 # if pod is terminating
+```
+
+In cases where `JobPodReplacementPolicy` and [PodFailurePolicy]((/docs/tasks/job/pod-failure-policy/)) is enabled, a Job will have a default (and only this value is allowed) value of `Failed`.
```

Comment:
Maybe we could also add a note to pod failure policy at the end, saying something like: 
```
Since Kubernetes v1.28, when pod failure policy is used, the Job controller will recreate
terminating pods only once they reach the terminal `Failed` phase. This behavior is analogous 
to when using `podRecreationPolicy: Failed`, see [pod recreation policy](#the link) for more details.
```

Reply:
Added.